src/wasm/wasm-binary.cpp

```diff
@@ -2686,12 +2686,11 @@ void WasmBinaryReader::readFunctions() {
 
     readVars();
 
-    std::swap(func->prologLocation, debugLocation);
+    func->prologLocation = debugLocation;
     {
       // process the function body
       BYN_TRACE("processing function: " << i << std::endl);
       nextLabel = 0;
-      debugLocation.clear();
```

Comment:
Why is this no longer needed?

Reply:
This change exposed a bug. If the first segment starts a bit before the beginning of the function, it may still cover some instructions. So, one should not clear the `debugLocation` here.

The `fn_prolog_epilog.debugInfo` test is failing without this change, since the debug information associated to the nop instruction at the beginning of the function is dropped.

Reply:
Which test do you mean by `fn_prolog_epilog.debugInfo`? I grepped for that name and didn't find anything.

Reply:
There is a file `test/fn_prolog_epilog.debugInfo.wasm`.